lib/AST/ASTPrinter.cpp

```diff
@@ -859,11 +859,16 @@ class PrintAST : public ASTVisitor<PrintAST> {
 
       // Get the innermost nominal type context.
       DeclContext *DC;
-      if (isa<NominalTypeDecl>(Current) || isa<ExtensionDecl>(Current))
+      if (isa<NominalTypeDecl>(Current))
         DC = Current->getInnermostDeclContext();
+      else if (isa<ExtensionDecl>(Current))
+        DC = Current->getInnermostDeclContext()->
+          getAsNominalTypeOrNominalTypeExtensionContext();
       else
         DC = Current->getDeclContext();
 
+      assert(DC->isTypeContext());
```

Comment:
Note that isTypeContext() is true for extension contexts, too.

So it looks like the original assert was this:

```
Assertion failed: (ownerNominal == baseTy->getAnyNominal()), function getContextSubstitutions, file /Volumes/ssd/SSDDocument/stash_master_sourceroot/swift/lib/AST/Type.cpp, line 3139.
```

What was ownerNominal and baseTy in that case? Is there a bug in getContextSubstitutions() when the DC is an extension of the base type? That should work...